### PR TITLE
fix user profile edit

### PR DESCRIPTION
**Pull request by onurccn**
- added cancel button on editing profile.
- added event handling for user fields explicitly (Departmant not added. Needs dropdown first!)

### CODE REVIEW

**Comment by onurccn** [comment]
[Profile page: Update user info is not working.](https://trello.com/c/nzNc6CDN/280-profile-page-update-user-info-is-not-working)

File: gtubt/lib/ui/pages/profile_page.dart [hunk 1]
```diff
@@ -476,14 +480,16 @@ class _ProfilePageState extends State<ProfilePage> {
                   NotificationFactory.errorFactory(message: state.errorMessage)
                       .show(context);
                 } else if (state.isLoading) {
-                  if (state.loadingMessage == null) {
+                  if (state.loadingMessage.isEmpty) {
                     context.read<AppbarBloc>().add(ShowLoading());
                   } else {
+                    // TODO: Can't dismiss shown notification explicitly.
```
Comment (gurolcay) [inline gtubt/lib/ui/pages/profile_page.dart]: could you check login form?

File: gtubt/lib/ui/pages/profile_page.dart [hunk 2]
```diff
@@ -242,6 +245,7 @@ class _ProfilePageState extends State<ProfilePage> {
   }
 
   Widget _phoneNumber(String? phone) {
+    // TODO: Needs custom formatting, we should use one of phone input widgets.
```
Comment (gurolcay) [inline gtubt/lib/ui/pages/profile_page.dart:250]: could you open issue card for this todo

File: gtubt/lib/service/user.dart [hunk 3]
```diff
@@ -4,10 +4,16 @@ import 'package:GTUBT/exceptions/user.dart';
 import 'package:GTUBT/models/api_response.dart';
 import 'package:GTUBT/models/user.dart';
 import 'package:GTUBT/service/service.dart';
+import 'dart:convert';
```
Comment (gurolcay) [inline gtubt/lib/service/user.dart]: please check import order.

Comment (gurolcay) [reply gtubt/lib/service/user.dart]: already imported

File: gtubt/lib/ui/pages/profile_page.dart (hunk 2 above)
Comment (onurccn) [reply gtubt/lib/ui/pages/profile_page.dart:250]: https://trello.com/c/cnFVbgRx/286-telefon-input-widget-i-eklenecek-profil-edit

**Review by gurolcay — APPROVED** [review]

**Review by aErgani — APPROVED** [review]

**Pull request merged**